CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -19,39 +19,61 @@ class TrackSoAHeterogeneousT {
+  constexpr auto chi2(int32_t i) const { return chi2_(i); }
+
+  // stateAtBS accessors
+  constexpr TrajectoryStateSoAT<S> & stateAtBS() { return stateAtBS_; }
+  constexpr TrajectoryStateSoAT<S> const & stateAtBS() const { return stateAtBS_; }
+  // eta accessors
+  constexpr auto & eta(int32_t i) { return eta_(i); }
+  constexpr auto eta(int32_t i) const { return eta_(i); }
+  // pt accessors
+  constexpr auto & pt(int32_t i) { return pt_(i); }
+  constexpr auto pt(int32_t i) const { return pt_(i); }
+
+  constexpr float charge(int32_t i) const { return std::copysign(1.f, stateAtBS_.state(i)(2)); }
+  constexpr float phi(int32_t i) const { return stateAtBS_.state(i)(0); }
+  constexpr float tip(int32_t i) const { return stateAtBS_.state(i)(1); }
+  constexpr float zip(int32_t i) const { return stateAtBS_.state(i)(4); }
+
+  // hitIndices accessors
+  constexpr auto & hitIndices() { return hitIndices_; }
+  constexpr auto const & hitIndices() const { return hitIndices_; }
+
+  // detInndices accessor
+  constexpr int nHits(int i) const { return detIndices_.size(i); }
+  constexpr auto & detIndices() { return detIndices_; }
+  constexpr auto const & detIndices() const { return detIndices_; }
+
+  // state at the detector of the outermost hit
+  // representation to be decided...
+  // not yet filled on GPU
+  // TrajectoryStateSoA<S> stateAtOuterDet;
+private:
   // Always check quality is at least loose!
   // CUDA does not support enums  in __lgc ...
-private:
   eigenSoA::ScalarSoA<uint8_t, S> quality_;
 
-public:
-  constexpr Quality quality(int32_t i) const { return (Quality)(quality_(i)); }
-  constexpr Quality &quality(int32_t i) { return (Quality &)(quality_(i)); }
-  constexpr Quality const *qualityData() const { return (Quality const *)(quality_.data()); }
-  constexpr Quality *qualityData() { return (Quality *)(quality_.data()); }
-
   // this is chi2/ndof as not necessarely all hits are used in the fit
-  eigenSoA::ScalarSoA<float, S> chi2;
-
-  constexpr int nHits(int i) const { return detIndices.size(i); }
+  eigenSoA::ScalarSoA<float, S> chi2_;
 
   // State at the Beam spot
   // phi,tip,1/pt,cotan(theta),zip
-  TrajectoryStateSoAT<S> stateAtBS;
-  eigenSoA::ScalarSoA<float, S> eta;
-  eigenSoA::ScalarSoA<float, S> pt;
-  constexpr float charge(int32_t i) const { return std::copysign(1.f, stateAtBS.state(i)(2)); }
-  constexpr float phi(int32_t i) const { return stateAtBS.state(i)(0); }
-  constexpr float tip(int32_t i) const { return stateAtBS.state(i)(1); }
-  constexpr float zip(int32_t i) const { return stateAtBS.state(i)(4); }
-
-  // state at the detector of the outermost hit
-  // representation to be decided...
-  // not yet filled on GPU
-  // TrajectoryStateSoA<S> stateAtOuterDet;
+  TrajectoryStateSoAT<S> stateAtBS_;
+  eigenSoA::ScalarSoA<float, S> eta_;
+  eigenSoA::ScalarSoA<float, S> pt_;
 
-  HitContainer hitIndices;
-  HitContainer detIndices;
+  HitContainer hitIndices_;
+  HitContainer detIndices_;
```

Comment:
I am not fully convinced of this encapsulations of SoAs.
It is just mere orthography. Semantics has not changed as both const and non-const accessors have been provided.
What are we encpsulating here.
The implementation encapsulation is done in the SoA itself.

Given that it is highly probable that all this will be modified again once the eventual final SoA model will be deployed (together with the Heterogeneous framework) I suggest to keep an open issue on Orthography of SoA
and leave for the time being my choice of orthography for my design of SoAs....

Reply:
So please split the changes in the Arithmetic of the fit (that is supposed (I hope) a performance improvement) from this esthetic changes. (also because we want to profile them independently)

Reply:
This commit just ensures we have a class with only one of each `public`, `protected` and `private` section. So indeed entirely cosmetic, but mandated by coding rules.

Computations and code style are already in different commits, so I guess you meant splitting this PR?

Reply:
yes, spitting this PR:
one cosmetics
one computational performance